app/models/course/discussion/topic.rb

```diff
@@ -3,4 +3,6 @@ class Course::Discussion::Topic < ActiveRecord::Base
 
   has_many :posts, dependent: :destroy, inverse_of: :topic
   has_many :subscriptions, dependent: :destroy, inverse_of: :topic
+
+  accepts_nested_attributes_for :posts
```

Comment:
Remember that if you are updating a topic you should not allow people to modify posts.